From a723585c5c465fcf3c216e962cbba60921cfc5cd Mon Sep 17 00:00:00 2001
From: Seokjae Byun <seokjae.byun@intel.com>
Date: Fri, 3 May 2013 10:09:48 -0700
Subject: [PORT FROM MAIN] wifi: Broadcom Hostapd ctrl connection issue

BZ: 98319

The ctrl_connection between Hostapd daemon and wifi HAL was not established for Broadcom solution.
So the code for android socket connection is added in "wpa_supplicant_8/hostapd/ctrl_iface.c"

Change-Id: I50a553c8636684c0f7ebaaac5f67031ccd3cc0e8
Orig-Change-Id: Ice4e99f7cc074a247da68d9609ac2020eacbc8c1
Category: device enablement
Domain: CWS.WIFI-Bluewoods
Origin: upstream
Upstream-Candidate: no,proprietary
Signed-off-by: Seokjae Byun <seokjae.byun@intel.com>
Signed-off-by: Jean-Michel Bachot <jean-michel.bachot@intel.com>
---
 hostapd/ctrl_iface.c |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/hostapd/ctrl_iface.c b/hostapd/ctrl_iface.c
index be941c4..c957eca 100644
--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -14,6 +14,10 @@
 #include <sys/stat.h>
 #include <stddef.h>
 
+#ifdef ANDROID
+#include <cutils/sockets.h>
+#endif /* ANDROID */
+
 #include "utils/common.h"
 #include "utils/eloop.h"
 #include "common/version.h"
@@ -1178,6 +1182,17 @@ int hostapd_ctrl_iface_init(struct hostapd_data *hapd)
 	if (hapd->conf->ctrl_interface == NULL)
 		return 0;
 
+#ifdef ANDROID
+        os_snprintf(addr.sun_path, sizeof(addr.sun_path), "wpa_%s",
+                    hapd->conf->ctrl_interface);
+        s = android_get_control_socket(addr.sun_path);
+
+        wpa_printf(MSG_DEBUG, "hostapd_ctrl_iface_init "
+                                   "addr.sun_path %s, s: %d", addr.sun_path, s );
+        if (s >= 0)
+                goto havesock;
+#endif /* ANDROID */
+
 	if (mkdir(hapd->conf->ctrl_interface, S_IRWXU | S_IRWXG) < 0) {
 		if (errno == EEXIST) {
 			wpa_printf(MSG_DEBUG, "Using existing control "
@@ -1288,6 +1303,10 @@ int hostapd_ctrl_iface_init(struct hostapd_data *hapd)
 	}
 	os_free(fname);
 
+#ifdef ANDROID
+havesock:
+#endif /* ANDROID */
+
 	hapd->ctrl_sock = s;
 	eloop_register_read_sock(s, hostapd_ctrl_iface_receive, hapd,
 				 NULL);
-- 
1.7.4.1

